src/core/dfn-abbr.js

```diff
@@ -0,0 +1,60 @@
+// @ts-check
+// Module core/dfn
+// - Finds all <dfn> elements with data-abbr attribute and parses them and registers definitions.
+// - Also populates abbrMap used by core/inlines for such definitions.
+
+import { registerDefinition } from "./dfn-map";
+
+export const name = "core/dfn-abbr";
+export const abbrMap = new Map();
+
+/**
+ * Generates simple Abbreviations from text passed.
+ *
+ * @param {String} text A string like "Permanent Account Number".
+ * @returns {String} abbr Abbreviation like "PAN"
+ */
+
+function getAbbreviationFromText(text) {
+  return text
+    .match(/\b([a-z])/gi)
+    .join("")
+    .toUpperCase();
+}
+
+/**
+ * Parses Elements of form <dfn data-abbr="PAN">Permanent Account Number</dfn>
+ * to <dfn data-abbr="PAN">Permanent Account Number (PAN)</dfn>
+ * @param {HTMLElement} dfn Element to be parsed.
+ */
+function renderAbbreviatedDefinition(dfn) {
+  // checks if text content is already in the form Permanent Account Number (PAN)
+  dfn.textContent = dfn.textContent.concat(` (${dfn.dataset.abbr})`);
```

Comment:
This might be overly destructive. Consider: 

```HTML
<dfn data-abbr>The <span>Funky</span> Bar</dfn>
```

Replacing the `textContent` could trash the `span`. The ` (${dfn.dataset.abbr})` should probably just be an appended TextNode

Reply:
```suggestion
  dfn.append(` (${dfn.dataset.abbr})`);
```

Reply:
I think we can just inline this anyway.

Reply:
Agree.